site/components/context/Transactions.js

```diff
@@ -0,0 +1,73 @@
+import { createContext, useCallback, useEffect, useState } from 'react'
+
+// @ts-ignore ts(2554)
+const TransactionsContext = createContext()
+
+export function TransactionsContextProvider({ children }) {
+  const [transactions, setTransactions] = useState([])
+  const [currentTransactions, setCurrentTransactions] = useState([])
+
+  const [modalIsOpen, setModalIsOpen] = useState(false)
+  const [openedOpId, setOpenedOpId] = useState(null)
+
+  const openModal = function ({ opId }) {
+    setOpenedOpId(opId)
+    setModalIsOpen(true)
+  }
+  const closeModal = () => setModalIsOpen(false)
+
+  useEffect(
+    function () {
+      if (!transactions.length) {
+        return
+      }
+      const groupedTransactions = transactions.reduce(
+        (result, transaction) => ({
+          ...result,
+          [transaction.opId]: result[transaction.opId]
+            ? { ...result[transaction.opId], ...transaction }
+            : transaction
+        }),
+        {}
+      )
+      setCurrentTransactions(Object.values(groupedTransactions))
+    },
+    [transactions]
+  )
+
+  useEffect(
+    function () {
+      if (!currentTransactions.length) {
+        return
+      }
+      openModal(currentTransactions[currentTransactions.length - 1])
+    },
+    [currentTransactions]
+  )
+
+  const addTransactionStatus = useCallback(
+    function (newTransaction) {
+      setTransactions((previousTransactions) => [
+        ...previousTransactions,
+        newTransaction
+      ])
+    },
+    [transactions]
+  )
+
+  return (
+    <TransactionsContext.Provider
+      value={{
+        addTransactionStatus,
+        closeModal,
+        currentTransactions,
+        modalIsOpen,
+        openedOpId
+      }}
+    >
+      {children}
+    </TransactionsContext.Provider>
+  )
+}
+
+export default TransactionsContext
```

Comment:
I think this file can be simplified. You might want to use the Vesper one as reference - see https://github.com/bloqpriv/vesper-monorepo/blob/master/app/components/TransactionContext.js

Of course, it might be a little bit of effort because the Vesper one has the funcitons to close and reopen multiple times, but it might be worth checking it - specifically, I think some of the functions exposed can be moved down to the Modal where they are used, and some of the effects can be dropped.

Reply:
Let's do this in a future PR.